packages/components/src/color-palette/test/index.tsx

```diff
@@ -153,12 +161,40 @@ describe( 'ColorPalette', () => {
 		expect( dropdownButton ).toBeVisible();
 
 		expect(
-			within( dropdownButton ).getByText( colors[ 0 ].name )
+			within( dropdownButton ).getByText( EXAMPLE_COLORS[ 0 ].name )
 		).toBeVisible();
 		expect(
 			within( dropdownButton ).getByText(
-				colors[ 0 ].color.replace( '#', '' )
+				EXAMPLE_COLORS[ 0 ].color.replace( '#', '' )
 			)
 		).toBeVisible();
 	} );
+
+	it( 'should show the clear button even when `colors` is an empty array', () => {
+		const onChange = jest.fn();
+
+		const { rerender } = render(
+			<ColorPalette
+				colors={ EXAMPLE_COLORS }
+				value={ INITIAL_COLOR }
+				onChange={ onChange }
+			/>
+		);
+
+		expect(
+			screen.getByRole( 'button', { name: 'Clear' } )
+		).toBeInTheDocument();
```

Comment:
Is this part of the test relevant? Won't it be the same if we remove it and just render once with `colors: []`?

Reply:
I thought it was relevant because we never check for the `Clear` button to be visible by default — do you think it would be better to split the check to a separate test?

Reply:
Okay, that makes sense. And I agree a separate test would be better - it would help express the intentions better. We don't seem to earn anything by re-rendering and staying in the same test.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/46001/commits/b6bae63553b3dac60b4e6fcd32fd718cea158511